test/lib/ansible_test/_internal/cloud/__init__.py

```diff
@@ -390,14 +423,26 @@ def _generate_resource_prefix():
         :rtype: str
         """
         if is_shippable():
-            return 'shippable-%s-%s' % (
+            prefix = 'ansible-test-shippable-%s-%s' % (
+                os.environ['SHIPPABLE_BUILD_NUMBER'],
+                os.environ['SHIPPABLE_JOB_NUMBER'],
+            )
+            short_prefix = 'ansible-test-%s-%s' % (
+                os.environ['SHIPPABLE_BUILD_NUMBER'],
+                os.environ['SHIPPABLE_JOB_NUMBER'],
+            )
+            unique = '%s-%s' % (
                 os.environ['SHIPPABLE_BUILD_NUMBER'],
                 os.environ['SHIPPABLE_JOB_NUMBER'],
             )
+        else:
+            unique = random.randint(10000000, 99999999)
+            node = re.sub(r'[^a-zA-Z0-9]+', '-', platform.node().split('.')[0]).lower()
 
-        node = re.sub(r'[^a-zA-Z0-9]+', '-', platform.node().split('.')[0]).lower()
+            prefix = 'ansible-test-%s-%d' % (node, unique)
+            short_prefix = 'ansible-test-%d' % (unique)
 
-        return 'ansible-test-%s-%d' % (node, random.randint(10000000, 99999999))
+        return prefix, short_prefix, unique
```

Comment:
Overall this looks good. However, I'm guessing the behavior of unique isn't what most users will expect.

These values are generated once before any test targets are executed. If there are multiple targets, or the same target is run more than once due to a retry, then the same "unique" prefix will be used multiple times.

To get a truly unique value for each test target, I would generate the non-random portion of the unique prefix here and then add the final random part from `random.randint` inside the `get_environment_config` function.